front-end/app/(tabs)/gallery.tsx

```diff
@@ -0,0 +1,530 @@
+      const validPhotos = photoList.filter(photo => {
+        if (!photo || !photo.id || !photo.uri || !photo.timestamp) {
+          return false;
+        }
+        
+        // On Android, only allow file:// URIs (local files)
+        if (Platform.OS === 'android') {
+          return photo.uri.startsWith('file://');
+        }
+        
+        // On iOS, allow all valid URIs
+        if (Platform.OS === 'ios') {
+          return photo.uri.startsWith('file://') || 
+                 photo.uri.startsWith('content://') || 
+                 photo.uri.startsWith('ph://');
+        }
+        
+        // On web, allow blob: URIs (web camera photos)
+        if (Platform.OS === 'web') {
+          return photo.uri.startsWith('blob:') || 
+                 photo.uri.startsWith('file://') ||
+                 photo.uri.startsWith('data:');
+        }
+        
+        // Default: allow all valid URIs
+        return true;
+      });
+
+      // Sort by timestamp (newest first)
+      validPhotos.sort((a, b) => b.timestamp - a.timestamp);
+      
+      // If we filtered out invalid photos, save the cleaned data
+      if (validPhotos.length < photoList.length) {
+        await AsyncStorage.setItem('inspecto_photos', JSON.stringify(validPhotos));
+      }
+
+      setPhotos(validPhotos);
+    } catch (error) {
+      console.error('Error loading photos:', error);
+      // If there's a persistent error, clear the corrupted data
+      try {
+        await AsyncStorage.removeItem('inspecto_photos');
+        await AsyncStorage.removeItem('last_gallery_visit');
+        await AsyncStorage.removeItem('gallery_seen_timestamp');
+      } catch (clearError) {
+      }
+      setPhotos([]);
+    } finally {
+      setLoading(false);
+    }
+  }
+
+
+  async function markGalleryAsSeen() {
+    try {
+      const currentTime = Date.now().toString();
+      await AsyncStorage.setItem('gallery_seen_timestamp', currentTime);
+    } catch (error) {
+    }
+  }
+
+  async function handleDeletePhoto(photoId: string) {
+    try {
+      // Only try MediaLibrary deletion on iOS
+      if (Platform.OS === 'ios') {
+        try {
+          await MediaLibrary.deleteAssetsAsync([photoId]);
+        } catch (error) {
+        }
+      }
+      
+      // Remove from AsyncStorage
+      const storedPhotos = await AsyncStorage.getItem('inspecto_photos');
+      if (storedPhotos) {
+        const photoList = JSON.parse(storedPhotos);
+        const updatedPhotos = photoList.filter((photo: Photo) => photo.id !== photoId);
+        await AsyncStorage.setItem('inspecto_photos', JSON.stringify(updatedPhotos));
+      }
+      
+      // Update local state
+      setPhotos(photos.filter(photo => photo.id !== photoId));
+      Alert.alert('Success', 'Photo deleted successfully.');
+    } catch (error) {
+      console.error('Error deleting photo:', error);
+      Alert.alert('Error', 'Failed to delete photo.');
+    }
+  }
+
+  function handlePhotoPress(photo: Photo) {
```

Comment:
for pressing photo, we actually want to view it in full screen rather than select to delete it. I think we should have a select icon at top to turn on select mode, when clicking it, then whichever photo u select it will be selected, but in normal mode, when you press the photo, it will just expand to fulll screen for annotation